packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -681,6 +682,27 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
         Definitions
       >
       const dispatch = useDispatch<ThunkDispatch<any, any, UnknownAction>>()
+      const internalStateRef = useRef<InternalMiddlewareState | null>(null)
```

Comment:
This logic in the hooks hile hurts a bit. Maybe the middleware could return a `hasSubscription` function instead of the internal state itself?